mk/dev.mk

```diff
@@ -13,7 +13,7 @@ SHELLCHECK_VERSION := v0.8.0
 YQ_VERSION := v4.24.2
 
 CI_KUBEBUILDER_VERSION ?= 2.3.2
-CI_KUBECTL_VERSION ?= v1.18.14
+CI_KUBECTL_VERSION ?= v1.23.5
```

Comment:
Does this affect the minimum k8s version we support?

Reply:
That's just a client, right? We have E2E for old K8S.

Reply:
Good point

Reply:
Yes, I think it's connected with this: https://kubernetes.io/releases/version-skew-policy/#kubectl also the policy changes and older versions might not support new versions of `kube-apiserver`. The tool version is (AFAIK) only used for unit tests.